x-pack/plugins/remote_clusters/server/routes/api/remote_clusters/register_delete_route.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { licensePreRoutingFactory } from '../../../lib/license_pre_routing_factory';
+import { isEsErrorFactory } from '../../../lib/is_es_error_factory';
+import { callWithRequestFactory } from '../../../lib/call_with_request_factory';
+import { wrapEsError, wrapUnknownError } from '../../../lib/error_wrappers';
+
+export function registerDeleteRoute(server) {
+  const isEsError = isEsErrorFactory(server);
+  const licensePreRouting = licensePreRoutingFactory(server);
+
+  server.route({
+    path: '/api/remote_clusters/{name}',
```

Comment:
Will this cause problems if the `name` has spaces or slashes in it? Does ES allow these characters in a cluster name?

Reply:
cluster name looks to be pretty restrictive, as they have to be able to be accessed via `cluster.remote.{CLUSTER_NAME}` on ES side. I tried with a couple of different special characters:

`test cluster` => `persistent setting [cluster.remote.test cluster.seeds], not recognized`
`test/cluster` => `persistent setting [cluster.remote.test/cluster.seeds], not recognized`
`test*cluster` => `persistent setting [cluster.remote.test*cluster.seeds], not recognized`

`_`, `-`, and alphanumeric characters appear to be allowed, so I think this route should be able to handle the cluster names fine 😄